plugins/cpu/fu-plugin-cpu.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2020 Mario Limonciello <mario.limonciello@dell.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-hash.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+gboolean
+fu_plugin_coldplug (FuPlugin *plugin, GError **error)
+{
+	gsize length;
+	g_autofree gchar *data = NULL;
+	g_auto(GStrv) lines = NULL;
+	g_autofree gchar *processor = NULL;
+	g_autofree gchar *model_name = NULL;
+	g_autofree gchar *microcode = NULL;
+	g_autofree gchar *vendor = NULL;
+	guint64 physical_id = G_MAXUINT64;
+
+	if (!g_file_get_contents ("/proc/cpuinfo", &data, &length, error))
+		return FALSE;
+
+	lines = g_strsplit (data, "\n", 0);
+	for (guint i = 0; lines[i] != NULL; i++) {
+		if (g_str_has_prefix (lines[i], "processor")) {
+			g_auto(GStrv) fields = g_strsplit (lines[i], ":", -1);
+			if (fields[1] != NULL) {
+				if (processor != NULL)
+					g_free (processor);
```

Comment:
are they going to be different? the "last" one wins? I think you could have `if (processor == NULL && g_str_has_prefix (lines[i], "processor")) {` and cut out a bit of code.

Reply:
Well you can't tell for sure how many CPUs are on the system in one pass of the for loop (and how many are going to be identical).  So I'm hoping this code will work for multi CPU package systems (make a device for each one) as well as multi CPU core systems (make one device).

Reply:
Although I guess I could reverse the loop from bottom of `/proc/cpuinfo` to get physical ID first.  If that didn't change don't bother reading all the other fields.